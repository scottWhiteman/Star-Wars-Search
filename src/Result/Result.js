import React from 'react';

export default class Result extends React.Component {
    formatRender() {
        switch (this.props.category) {
            case 'people':
                return this.formatPerson();
            case 'species':
                return this.formatSpecies();
            case 'films':
                return this.formatFilm();
            case 'planets':
                return this.formatPlanet();
            case 'vehicles':
                return this.formatVehicle();
            case 'starships':
                return this.formatShip();
            default:
                return <h2>That Category does not exist!</h2>
        }
    }
    
    formatPerson = () => {
        return (
            <>
                <h2>{this.props.item.name}</h2>
                <p>{this.props.item.species}</p>
                <p>{this.props.item.homeworld}</p>
            </>
        );
    }

    formatSpecies = () => {
        return (
            <>
                <h2>{this.props.item.name}</h2>
                <p>{this.props.item.classification}</p>
                <p>{this.props.item.designation}</p>
                <p>{this.props.item.homeworld}</p>
            </>
        );
    }

    formatFilm = () => {
        return (
            <>
                <h2>{this.props.item.title}</h2>
                <p>{this.props.item.created}</p>
                <p>{this.props.item.director}</p>
            </>
        );
    }

    formatPlanet = () => {
        return (
            <>
                <h2>{this.props.item.name}</h2>
                <p>{this.props.item.diameter}</p>
                <p>{this.props.item.population}</p>
            </>
        );
    }

    formatVehicle = () => {
        return (
            <>
                <h2>{this.props.item.name}</h2>
                <p>{this.props.item.model}</p>
                <p>{this.props.item.vehicle_class}</p>
            </>
        );
    }

    formatShip = () => {
        return (
            <>
                <h2>{this.props.item.name}</h2>
                <p>{this.props.item.model}</p>
                <p>{this.props.item.length}</p>
                <p>{this.props.item.hyperdrive_rating}</p>
            </>
        );
    }

    render() {
        return (
            <div className="Result-Container">
                {this.formatRender()}
            </div>
        );
    }
}