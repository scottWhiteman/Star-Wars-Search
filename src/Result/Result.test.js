import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Result from './Result';

describe("Result componenet test", () => {
    it('renders Result without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Result />, div)
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders Result to UI without data', () => {
        const tree = renderer.create(<Result />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    const item = {
        name: "Darth",
        species: "Helmet",
        homeworld: "Space"
    }
    const category = "people";

    it('renders Result to UI with data', () => {
        const tree = renderer.create(<Result item={item} category={category}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})