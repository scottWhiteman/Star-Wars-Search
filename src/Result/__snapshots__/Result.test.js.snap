// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result componenet test renders Result to UI 1`] = `
<div
  className="Result-Container"
>
  <h2>
    That Category does not exist!
  </h2>
</div>
`;

exports[`Result componenet test renders Result to UI 2`] = `
<div
  className="Result-Container"
>
  <h2>
    Darth
  </h2>
  <p>
    Helmet
  </p>
  <p>
    Space
  </p>
</div>
`;

exports[`Result componenet test renders Result to UI with data 1`] = `
<div
  className="Result-Container"
>
  <h2>
    Darth
  </h2>
  <p>
    Helmet
  </p>
  <p>
    Space
  </p>
</div>
`;

exports[`Result componenet test renders Result to UI without data 1`] = `
<div
  className="Result-Container"
>
  <h2>
    That Category does not exist!
  </h2>
</div>
`;
